### ISSUE TITLE
How to realize manual upgrade of meshcentral

### ISSUE THREAD

**Issue by billydu1980** [Ylianst/MeshCentral#2510]
Dear Ylianst‘s team:

We have a challenge. In a factory LAN environment, the version of meshCentral  is 0.4.2-x. Host Linux is CentOS 7.6.
There are more than 50 connected agent PCs, most of them are win7 (32bit) systems, a few agentless AMT connections.
We want to manually upgrade meshcentral to 0.7.65.
We need to keep all the original agent connection information. How to operate? Which files in meshcentral-data（ 0.4.2-x） need to continue to be used.
Nodejs version is node-v8.11. X and has been upgraded to v12.8. X.
Mongodb has been upgraded to 4.4.1.

Thanks so much!
BillyDu

**Comment by nzalev** [comment]
I have always performed meshcentral version upgrades in-place.

My procedure is as follows (on unix):

1. Stop meshcentral
2. Create a backup of the entire meshcentral directory (for roll-back purposes)
2. Switch into the user that runs meshcentral (for me this is _meshcentral)
3. cd into the meshcentral directory
4. run `npm install meshcentral@desired_version` (in your case `npm install meshcentral@0.7.65`)
5. Start meshcentral

**Comment by Mfinadmin** [comment]
> I have always performed meshcentral version upgrades in-place.
> 
> My procedure is as follows (on unix):
> 
> 1. Stop meshcentral
> 2. Create a backup of the entire meshcentral directory (for roll-back purposes)
> 3. Switch into the user that runs meshcentral (for me this is _meshcentral)
> 4. cd into the meshcentral directory
> 5. run `npm install meshcentral@desired_version` (in your case `npm install meshcentral@0.7.65`)
> 6. Start meshcentral

I am also doing Procedure same for Windows...

**Comment by OutbackMatt** [comment]
In the Mesh Admin GUI (mesh.example.com), on the 'My Server' page (bottom tab on the left), there is a 'check server version' link that will show you the current version and the upgrade options.

Upgrade by clicking on the link.

All automated in the background, takes about 5-10 seconds on my machine

is there a specific reason that you want to go to 0.7.65
Latest stable is shown as 0.7.89 currently

**Comment by Ylianst** [comment]
Yes, as indicated by @OutbackMatt, automatic update is done int he MyServe tab here:

![image](https://user-images.githubusercontent.com/1319013/114923606-e2034680-9de1-11eb-9a11-689ba526659e.png)

Manual update exactly as indicated by @nzalev.

**Comment by billydu1980** [comment]
Sorry, maybe I didn't make it clear。

Then meshcentral server int the factory environment cannot be connected to the internet network,

I have downloaded and installed version 0.7.89 and new meshcentral files on other computers with NPM. I want to replace the old file with the new files(node_ module).

In order to keep all the connection information of agent and AMT, which old files in the meshcentral-data need to be kept in use.

BillyDu

**Comment by Ylianst** [comment]
I see.

- Install MeshCentral on a computer that does have access to the internet.
- Put in a config.json that is similar to the one you use in production.
- Run MeshCentral once using `node node_modules/meshcentral`, this may install extra modules.
- Stop production server.
- On production server, rename "node_modules" to "node_modules_old"
- Copy "node_modules" folder from Internet computer to production computer.
- Start production server.

Basically, you just need to stop and swap the "node_modules" folder and start again. You will have the latest update. As always, make sure to always backup your "meshcentral-data" and database offsite. Hope that helps.

**Comment by billydu1980** [comment]
Dear Ylianst，
Thank you very much. I have successfully completed the test.
Offline upgrade is very meaningful in industrial occasions.

BillyDU
